lib/module.ts

```diff
@@ -86,21 +112,16 @@ export class Module<
   }
 
   private registerOnDataChangeHooks(state: UnwrapNestedRefs<S>): void {
-    const onDataChangeHooks = this.options.plugins
+    const onDataChangeHooks = this.plugins
       ?.filter((plugin) => plugin.onDataChange !== undefined)
       .map((plugin) => plugin.onDataChange)
     if (onDataChangeHooks?.length) {
-      watch(
-        state,
-        (value, oldValue, onCleanup) => {
-          onDataChangeHooks.forEach((callback) =>
-            callback?.(value, oldValue as NonNullable<typeof oldValue>, onCleanup),
-          )
-        },
-        {
-          immediate: true,
```

Comment:
Removed the immediate call, I thought it might be surprising to users. I think the state init callback is sufficient for doing things when the module state is created.